pybamm/simulation.py

```diff
@@ -211,39 +211,84 @@ def set_up_experiment(self, model, experiment):
                     "Voltage input [V]": 0,  # doesn't matter
                     "Power input [W]": P,
                 }
-            # Update period
-            operating_inputs["period"] = op[3]
-            # Update events
-            if events is None:
-                # make current and voltage values that won't be hit
-                operating_inputs.update(
-                    {"Current cut-off [A]": -1e10, "Voltage cut-off [V]": -1e10}
-                )
-            elif events[1] in ["A", "C"]:
-                # update current cut-off, make voltage a value that won't be hit
-                if events[1] == "A":
-                    I = events[0]
-                else:
-                    # Scale C-rate with capacity to obtain current
-                    capacity = self._parameter_values["Cell capacity [A.h]"]
-                    I = events[0] * capacity
-                operating_inputs.update(
-                    {"Current cut-off [A]": I, "Voltage cut-off [V]": -1e10}
-                )
-            elif events[1] == "V":
-                # update voltage cut-off, make current a value that won't be hit
-                V = events[0]
-                operating_inputs.update(
-                    {"Current cut-off [A]": -1e10, "Voltage cut-off [V]": V}
-                )
-
-            self._experiment_inputs.append(operating_inputs)
-            # Add time to the experiment times
-            dt = op[2]
-            if dt is None:
-                # max simulation time: 1 week
-                dt = 7 * 24 * 3600
-            self._experiment_times.append(dt)
+            elif op[1] == "Drive":
+                # Update inputs for Drive Cycle
+                driving_inputs_list = []
+                for I_drive in op[0]:
+                    # Step though all drive cycle inputs and
+                    # generate list containing driving_inputs for each step
+                    driving_inputs = {
+                        "Current switch": 1,
+                        "Voltage switch": 0,
+                        "Power switch": 0,
+                        "Current input [A]": I_drive,
+                        "Voltage input [V]": 0,  # doesn't matter
+                        "Power input [W]": 0,  # doesn't matter
+                    }
+                    # Update period
+                    driving_inputs["period"] = op[3]
```

Comment:
here period should also be a list, which is updated based on the times in the drive cycle